docs/docs/concepts/foundation/communication/public_private_calls/slow_updates_tree.md

```diff
@@ -0,0 +1,97 @@
+
+## How it works
+
+We developed the Slow Update Tree to help balance public and private execution in a blockchain context.
+
+Earlier systems typically used either fully public or entirely private state trees, which led to privacy leaks if reading public data from private state.
+
+Using a shared state tree, it is possible to privately access historic public data by providing a membership proof to show that a value is indeed part of a commitment, and then check that the commitment matches the one stored in the state, as shown below.
+
+```mermaid
+graph TD;
+    Cm[CommitmentToKnowingAValue] --> Vm1[Value1]
+    Cm --> Vm2[Value2]
+    Cm --> Vmn[Value3]
+    Vm1 --> Mp1[Membership Proof for Value1]
+    Mp1 --> St[Commitment in State]
+    St --> Cm
+```
+
+However this means that any changes to the commitment will invalidate our membership proof, and make our read fail.
+
+To solve this, the Slow Update Tree is a dual-tree structure - using a current and pending tree - that updates at the end of an epoch. This way, we can ensure that the values are stable throughout the epoch, and that the membership proofs are not invalidated by changes in other contracts more than once every epoch.
+
+```mermaid
+graph TD;
+    Change{Epoch Over} -->|True| Current{Current}
+    Change -->|False| Pending{Pending}
+    Current --> Current1[Current Commitment 1]
+    Current --> CurrentM[Current Commitment 2]
+    CurrentM --> Value1[Current Value 1]
+    CurrentM --> Value2[Current Value 2]
+    CurrentM --> ValueN[Current Value n]
+    Pending --> PendingM[Pending Commitment 1]
+    PendingM --> PValue1[Pending Value 1]
+    PendingM --> PValue2[Pending Value 2]
+    PendingM --> PValueN[Pending Value n]
+```
+
+### Dual Tree Structure
+
+The Slow Update Tree operates on a dual-tree mechanism - a _current_ tree and a _pending_ tree. These trees manage the state data that needs to be shared between public and private executions.
+
+This _slow update_ mechanism minimizes the invalidation of reads and writes due to concurrent updates.
+
+### Epoch-Based Updates
+
+The current tree is replaced with the pending tree at the end of each epoch. Then a new pending tree is created, the epoch restarts, and the cycle begins again.
+
+### Accessing Data
+
+**From public state:** Accessed directly from the state
+**From private state:** Performs a membership proof for the values in the tree, ensuring that they are part of the commitment.
+
+### Updating Values
+
+Updates are made to the pending tree. Then at the end of each epoch, the updates in the pending tree are committed and it becomes the current tree.
+
+## Limitations
+
+### Delayed State Finality
+
+Updates in the Slow Update Tree are only finalized at the end of an epoch.
+
+This could potentially lead to delays in state changes - for example an address may be added to a blacklist but this will not be reflected in the state until the end of the current epoch.
```

Comment:
I wouldn't call this a limitation but more like a feature. The tree is only useful for historic data for a reason